docs/dev/design/better-doc-urls-handling.rst

```diff
@@ -392,6 +408,95 @@ this allow us to serve subprojects without using a prefix.
      - project
      - The subproject/file doesn't exist
 
+Code example
+------------
+
+This is a simplified version of the implementation.
+
+.. code-block:: python
+
+   from readthedocs.projects.models import Project
+
+   LANGUAGES = {"es", "en"}
+
+
+   def resolve(canonical_project: Project, path_parts: list[str], check_subprojects=True):
+      # Multiversion project.
+      language = path_parts[0]
+      version_slug = path_parts[1]
+      if not canonical_project.single_version and language in LANGUAGES:
+         if canonical_project.language == language:
+               project = canonical_project
+         else:
+               project = canonical_project.translations.filter(language=language).first()
+         if project:
+               version = project.versions.filter(slug=version_slug).first()
+               if version:
+                  return project, version, "/".join(path_parts[2:])
+               return project, None, None
+
+      # Subprojects.
+      prefix = path_parts[0]
+      project_slug = path_parts[1]
+      if check_subprojects and prefix == "projects":
+         project = canonical_project.subprojects.filter(alias=project_slug).first()
+         if project:
+               return resolve(
+                  canonical_project=project,
+                  path_parts=path_parts[2:],
+                  check_subprojects=False,
+               )
```

Comment:
Is this calling itself in a recursive way or it's calling the current `resolve` function we have now?

Reply:
this is  recursive, yes, guess this mimics more what we currently call `unresolve`, and what we call `resolve` is more like a `reverse` (from django.urls.reverse).